### PR TITLE
Added unit tests to the project

### PR DESCRIPTION
**Pull request by dmccoystephenson**
## Problem
There are no unit tests implemented for this project.

## Solution
Unit tests have been implemented.

### New Classes
In order to make the project more testable, the following classes have been introduced to wrap certain operations:
- HttpClientFactory
- HttpEntityStringifier
- KeyStoreReader
- RestTemplateFactory
- SSLContextFactory

Additionally, the Message class has been moved to its own file.

### Typo Correction
The 'certficates' typo has been corrected to 'certificates' across the board.

## Testing
Unit tests have been verified to pass & the signing functionality has been verified to work.

### CODE REVIEW

**Review by drewjj — CHANGES_REQUESTED** [review]
Everything is looking great! These are some great additions for this repository and everything runs perfectly while running the tests through a command line. I did encounter an issue with the ApplicationTest test() test in VSCode. It appears JMockit is not being properly initialized in the Java TestRunner. This might be due to the dev container configuration but it would be good to get this resolved since we do want to include that in this pull request.

**Review by drewjj — APPROVED** [review]
Switching to an approval with comments. My previous comment is still my only request. Might be worth to make a backlog item for it.

File: src/main/java/us/dot/its/jpo/sec/helpers/HttpClientFactory.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package us.dot.its.jpo.sec.helpers;
```
Comment (payneBrandon) [inline src/main/java/us/dot/its/jpo/sec/helpers/HttpClientFactory.java:1]: Looks like we abstracted some of this functionality into helper classes, but only have unit tests covering one of the new classes. Do we need to add additional tests?

File: test-scripts/hit_endpoint.sh [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+curl -X POST localhost:8090/sign -H "Content-Type: application/json" -d "{\"message\": \"test\", \"sigValidityOverride\": 0}"
```
Comment (payneBrandon) [inline test-scripts/hit_endpoint.sh:3]: Do we need to document this script somewhere to show what it's for?

**Review by payneBrandon — CHANGES_REQUESTED** [review]
Just a couple questions

File: src/main/java/us/dot/its/jpo/sec/helpers/HttpClientFactory.java (hunk 1 above)
Comment (dmccoystephenson) [reply src/main/java/us/dot/its/jpo/sec/helpers/HttpClientFactory.java:1]: That seems reasonable. Unit tests have been added for all helper classes.

File: test-scripts/hit_endpoint.sh (hunk 2 above)
Comment (dmccoystephenson) [reply test-scripts/hit_endpoint.sh:3]: A test-scripts README has been added with documentation for this script.

**Review by payneBrandon — APPROVED** [review]
lgtm!

**Pull request merged**